assets/src/block-editor/components/with-deprecation-notice.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+
+export default createHigherOrderComponent(
```

Comment:
See my other comment. I would apply `withSelect` with a JS version of `ampBlocksInUse` here.